File: Java/AndresP/finpair.java
 
 public class JavaApplication20 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a[]={1,2,3,4,5,7,8};
        int missed=getMisigno(a, a.length);
        System.out.println("numero perdido"+missed);
       
       
    }
 static boolean  finpair(int arr[],int x){
        int size=arr.length;
        int i=0,j=1;
        while (i<size&&j<size) {            
            if (i!=j&&arr[i]==x) {
                System.out.println("par encontrado: "+"("+arr[i]+"-"+arr[j]+")");
                return true;
            }
            else if(arr[j]-arr[i]<x){
                j++;               
            }else{
                j++;
            }
            
        }
        System.out.println("no existe el par q debe ser igual a n al rstar");
        return false;
    }
}

File: Codigos estudiantes por lenguaje/PY/andresp/finpair.py
def finpair(arr,x):
    size=len(arr)
    i,j=0,1
    while(i<size and j<size):
        if(i!=j and arr[j] - arr[i]==x):
            print("par encontrado: "+"(",arr[j],"-",arr[i],")")
            return True
        elif (arr[j]-arr[i]<x):

            j += 1
        else:
            i += 1
    print("no existe el par q debe ser igual a n al rstar")
    return False
a=[1,3,4,5,8,10,11]
x=6
finpair(a,x)
